fractal_server/app/api/v1/job.py

```diff
@@ -46,3 +47,46 @@ async def get_job(
         pass
 
     return job_read
+
+
+@router.get("/download/{job_id}", response_class=StreamingResponse)
+async def download_job_logs(
+    job_id: int,
+    user: User = Depends(current_active_user),
+    db: AsyncSession = Depends(get_db),
+) -> Optional[ApplyWorkflow]:
+    job = await db.get(ApplyWorkflow, job_id)
+    if not job:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail="Job not found"
+        )
+    await get_project_check_owner(
+        project_id=job.project_id, user_id=user.id, db=db
+    )
+
+    job_read = ApplyWorkflowRead(**job.dict())
+
+    from devtools import debug
+
+    debug(job_read)
+
+    from io import BytesIO
+    from zipfile import ZipFile
+    from zipfile import ZIP_DEFLATED
+    import os
+
+    io = BytesIO()
+    zip_sub_dir = (
+        f"{job_read.project_id}_{job_read.workflow_id}_{job_id}_archive"
+    )
+    zip_filename = "%s.zip" % zip_sub_dir
+    with ZipFile(io, mode="w", compression=ZIP_DEFLATED) as zip:
+        for fpath in os.listdir(job_read.working_dir):
+            zip.write(job_read.working_dir / fpath)
+        # close zip
+        zip.close()
```

Comment:
Already taken care of by `with Zipfile() as zip`?

Reply:
I take it back.
But then let's use an instance rather than `with`

Reply:
> ZipFile is also a context manager and therefore supports the [with](https://docs.python.org/3/reference/compound_stmts.html#with) statement. ...closed after the with statement’s suite is finished—even if an exception occurs...

my bad, it's already a context manager. Just remove .close() method